packages/flutter_tools/test/general.shard/dart_plugin_test.dart

```diff
@@ -772,6 +772,130 @@ void main() {
             '\n\n');
       });
 
+      testUsingContext('provides warning when a plugin references a default plugin without implementation', () async {
+        final Set<String> directDependencies = <String>{'url_launcher'};
+        final List<PluginInterfaceResolution> resolutions =
+            resolvePlatformImplementation(
+          <Plugin>[
+            Plugin.fromYaml(
+              'url_launcher',
+              '',
+              YamlMap.wrap(<String, dynamic>{
+                'platforms': <String, dynamic>{
+                  'linux': <String, dynamic>{
+                    'default_package': 'url_launcher_linux',
+                  },
+                },
+              }),
+              null,
+              <String>[],
+              fileSystem: fs,
+              appDependencies: directDependencies,
+            ),
+            Plugin.fromYaml(
+              'url_launcher_linux',
+              '',
+              YamlMap.wrap(<String, dynamic>{
+                'implements': 'url_launcher',
+                'platforms': <String, dynamic>{},
+              }),
+              null,
+              <String>[],
+              fileSystem: fs,
+              appDependencies: directDependencies,
+            ),
+          ],
+          selectDartPlugins: true,
+        );
+
+        expect(resolutions.length, equals(0));
+        expect(
+            testLogger.warningText,
+            'Package url_launcher:linux references url_launcher_linux:linux as the default plugin, '
+            'but it does not provide an inline implementation.\n'
+            'Ask the maintainers of url_launcher to either avoid referencing a default implementation via `platforms: linux: default_package: url_launcher_linux` '
+            'or add an inline implementation to url_launcher_linux via `platforms: linux: [dart]PluginClass`.\n'
+            '\n');
+      });
+
+      testUsingContext('avoid warning when a plugin references a default plugin with a native implementation only', () async {
```

Comment:
Is there a corresponding test that there's no warning when there's only a Dart implementation, and no native implementation? That's a valid case as well.

Reply:
I added the test in 9c12ec84693fc56b93604b16275102cabcb32c76, but it fails, as I expect no native plugin is resolved, which is a false assumption. More [here](https://github.com/flutter/flutter/pull/137040#issuecomment-2178501864)